src/core/Akka/Delivery/Internal/ProducerControllerImpl.cs

```diff
@@ -490,10 +490,10 @@ private void ReceiveRequest(long newConfirmedSeqNr, long newRequestedSeqNr, bool
             : newRequestedSeqNr;
 
         if (newRequestedSeqNr2 != newRequestedSeqNr)
-            _log.Debug("Expanded requestedSeqNr from [{0}] to [{1}], because current [{3}] and all were probably lost.",
+            _log.Debug("Expanded requestedSeqNr from [{0}] to [{1}], because current [{2}] and all were probably lost.",
                 newRequestedSeqNr, newRequestedSeqNr2, stateAfterAck.CurrentSeqNr);
 
-        if (newRequestedSeqNr > CurrentState.RequestedSeqNr)
+        if (newRequestedSeqNr2 > CurrentState.RequestedSeqNr)
```

Comment:
This was the bug - we need to evaluate `newRequestedSeqNr2` here for when the producer believes that the consumer has lost data (and re-delivery isn't enabled). This is what was causing this test to get stuck and probably some of the others in this suite.